File: Task 5/Java/app/src/main/java/dev/ogabek/java/network/services/EmployeeService.java
package dev.ogabek.java.network.services;

import dev.ogabek.java.model.Employee;
import dev.ogabek.java.model.Respond;
import dev.ogabek.java.model.RespondDelete;
import dev.ogabek.java.model.RespondUpdate;
import retrofit2.Call;
import retrofit2.http.*;

public interface EmployeeService {

    @GET("employees")
    Call<Respond> get();

    @PUT("update/{id}")
    Call<RespondUpdate> put(@Path("id") int id, @Body Employee employee);

    @POST("create")
    Call<RespondUpdate> post(@Body Employee employee);

    @DELETE("delete/{id}")
    Call<RespondDelete> delete(@Path("id") int id);

}

File: Task 5/Kotlin/app/src/main/java/dev/ogabek/kotlin/network/service/EmployeeService.kt
package dev.ogabek.kotlin.network.service

import dev.ogabek.kotlin.model.Employee
import dev.ogabek.kotlin.model.Respond
import dev.ogabek.kotlin.model.RespondDelete
import dev.ogabek.kotlin.model.RespondUpdate
import retrofit2.Call
import retrofit2.http.*

interface EmployeeService {

    @GET("employees")
    fun get(): Call<Respond>

    @POST("create")
    fun post(@Body employee: Employee): Call<RespondUpdate>

    @PUT("update/{id}")
    fun put(@Path("id") id: Int, @Body employee: Employee): Call<RespondUpdate>

    @DELETE("delete/{id}")
    fun delete(@Path("id") id: Int): Call<RespondDelete>

}